src/legacy/core_plugins/vis_default_editor/public/components/controls/time_interval.tsx

```diff
@@ -56,15 +58,45 @@ function TimeIntervalParamEditor({
   let selectedOptions: ComboBoxOption[] = [];
   let definedOption: ComboBoxOption | undefined;
   let isValid = false;
+  const errors = [];
   if (value) {
     definedOption = find(options, { key: value });
     selectedOptions = definedOption ? [definedOption] : [{ label: value, key: 'custom' }];
-    isValid = !!(definedOption || isValidInterval(value, timeBase));
+    isValid = !!definedOption || isValidInterval(value, timeBase);
+
+    if (!isValid) {
+      errors.push(
+        i18n.translate('visDefaultEditor.controls.timeInterval.invalidFormatErrorMessage', {
+          defaultMessage: 'Invalid interval format.',
+        })
+      );
+    }
+  }
+
+  let interval: { scaled: boolean; scale: number; expression: string } = {} as any;
+  if (isValid) {
+    interval = get(agg, 'buckets.getInterval') && (agg as any).buckets.getInterval();
+
+    // we check if Elasticsearch interval is valid to show a user appropriate error message
+    // we don't check it for 0ms because the overall time range has not yet been set
+    const isValidEs = interval.expression === '0ms' ? true : isValidEsInterval(interval.expression);
```

Comment:
The check for `0ms` comes from interval's write method in `date_histogram.ts`:
https://github.com/elastic/kibana/blob/01571b67394552da3c73e9dbdf75eaca245d3aff/src/plugins/data/public/search/aggs/buckets/date_histogram.ts#L181